File: rules-interpreter/src/main/java/io/github/iamrenny/ruleflow/evaluators/BinaryOrContextEvaluator.java
package io.github.iamrenny.ruleflow.evaluators;

import io.github.iamrenny.ruleflow.RuleFlowLanguageParser;
import io.github.iamrenny.ruleflow.visitors.Visitor;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class BinaryOrContextEvaluator implements ContextEvaluator<RuleFlowLanguageParser.BinaryOrContext> {
    private static final Logger logger = LoggerFactory.getLogger(BinaryOrContextEvaluator.class);

    @Override
    public Boolean evaluate(RuleFlowLanguageParser.BinaryOrContext ctx, Visitor visitor) {
        boolean res = (Boolean) visitor.visit(ctx.left) || (Boolean) visitor.visit(ctx.right);
        logger.debug("BinaryOr: left={}, right={} result={}", visitor.visit(ctx.left), visitor.visit(ctx.right), res);
        return res;
    }
}

File: rules-interpreter/src/main/kotlin/com/rappi/fraud/rules/parser/evaluators/BinaryOrContextEvaluator.kt
package com.rappi.fraud.rules.parser.evaluators

import com.rappi.fraud.analang.ANALexer
import com.rappi.fraud.analang.ANAParser
import com.rappi.fraud.rules.parser.visitors.Visitor

class BinaryOrContextEvaluator : ContextEvaluator<ANAParser.BinaryOrContext> {

    override fun evaluate(ctx: ANAParser.BinaryOrContext, visitor: Visitor): Boolean {
        return visitor.visit(ctx.left) as Boolean || visitor.visit(ctx.right) as Boolean
    }
}